src/router_core/connections.c

```diff
@@ -1381,6 +1381,10 @@ static void qdr_connection_opened_CT(qdr_core_t *core, qdr_action_t *action, boo
         qdrc_event_conn_raise(core, QDRC_EVENT_CONN_OPENED, conn);
     }
 
+    if (discard && conn) {
```

Comment:
Code nit - why stuff this way down on the tail?   Why not do an immediate check for discard at the start of the function?  That would be consistent with how the discard flag is handled w.r.t. other action routines:
if (discard) {
   if (conn) qd_connection_free(conn);
   return;
}

Added bonus - drops a whole level of indentation.

Reply:
Thanks @kgiusti I have added code to check conn and discard at the top of the functions like other action routines and also dropped a level of indentation. The diff looks weird because of dropping one level of  indentation